isis/src/control/apps/findfeatures/GenericTiledTransform.cpp

```diff
@@ -0,0 +1,180 @@
+  setMatrix(cv::Mat::eye(3, 3, CV_64FC1));
+  setSize(cv::Size(0,0));
+}
+
+/** Named generic identity matrix */
+GenericTiledTransform::GenericTiledTransform(const QString &name, const int tileSize) : GenericTransform(name), m_tileSize(tileSize) {
+  setMatrix(cv::Mat::eye(3, 3, CV_64FC1));
+  setSize(cv::Size(0,0));
+}
+
+/** Construct named transform with a 3x3 transformation matrix and a tilesize*/
+GenericTiledTransform::GenericTiledTransform(const QString &name, const cv::Mat &matrix, const int tileSize) :
+                                   GenericTransform(name), m_tileSize(tileSize){
+  setMatrix(matrix);
+  setSize(cv::Size(0,0));
+}
+
+/** Construct named transfrom with 3x3 matrix with a size specification and a tilesize*/
+GenericTiledTransform::GenericTiledTransform(const QString &name, const cv::Mat &matrix,
+                                   const cv::Size &size,
+                                   const int tileSize) :
+                                   GenericTransform(name), m_tileSize(tileSize){
+  setMatrix(matrix);
+  setSize(size);
+}
+
+/** Construct named transfrom with 3x3 matrix with a subarea specification and tilesize */
+GenericTiledTransform::GenericTiledTransform(const QString &name, const cv::Mat &matrix,
+                                   const GenericTransform::RectArea &subarea,
+                                   const int tileSize) :
+                                   GenericTransform(name), m_tileSize(tileSize) {
+  cv::Mat tmatrix = ImageTransform::translation(-subarea.x, -subarea.y);
+  setMatrix( tmatrix * matrix );
+  setSize( subarea.size() );
+}
+
+int GenericTiledTransform::getTileSize() const{
+  return m_tileSize;
+}
+
+GenericTiledTransform::RectArea GenericTiledTransform::transformedSize(const cv::Mat &tmat,
+                                                        const cv::Size &imSize) {
+
+  std::vector<cv::Point2f> t_corners;
+  cv::perspectiveTransform(corners(imSize), t_corners, tmat);
+
+  double xmin(t_corners[0].x), xmax(t_corners[0].x);
+  double ymin(t_corners[0].y), ymax(t_corners[0].y);
+  for (unsigned int i = 1 ; i < t_corners.size() ; i++) {
+    if ( t_corners[i].x < xmin) xmin = t_corners[i].x;
+    if ( t_corners[i].x > xmax) xmax = t_corners[i].x;
+    if ( t_corners[i].y < ymin) ymin = t_corners[i].y;
+    if ( t_corners[i].y > ymax) ymax = t_corners[i].y;
+  }
+
+  return ( RectArea(xmin, ymin, (int) (xmax-xmin+0.5), (int) (ymax-ymin+0.5)) );
+}
+
+std::tuple<int, int> GenericTiledTransform::getNumTiles(const cv::Size size) const{
+  int width = size.width;
+  int height = size.height;
+  int tileSize = getTileSize();
+  int nxTiles = (width + tileSize - 1) / tileSize;
+  int nyTiles = (height + tileSize - 1) / tileSize;
+  return {nxTiles, nyTiles};
+}
+
+std::pair<cv::Range, cv::Range> GenericTiledTransform::getTile(int tileID, cv::Size size) const{
```

Comment:
Particularly here it would be helpful to document that tileID increases in the X direction first. So, a 2 by 4 would have these IDs:

```
0 1 2 3
4 5 6 7
```